src/critics/path_angle_critic.cpp

```diff
@@ -28,16 +38,34 @@ void PathAngleCritic::score(models::CriticFunctionData & data)
     return;
   }
 
-  auto goal_x = xt::view(data.path, -1, 0);
-  auto goal_y = xt::view(data.path, -1, 1);
+  utils::setPathFurthestPointIfNotSet(data);
+
+  auto offseted_idx = std::min(
+    *data.furthest_reached_path_point + offset_from_furthest_, data.path.shape(0) - 1);
+
+  double goal_x = xt::view(data.path, offseted_idx, 0);
+  double goal_y = xt::view(data.path, offseted_idx, 1);
+
+  bool angle_to_furthest_more_than_threshold = utils::pose_point_angle(
+    data.state.pose.pose, goal_x,
+    goal_y) >
```

Comment:
Line break here makes it difficult to read

Reply:
resolved